src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -715,7 +741,7 @@ private int decodePublish(
             else
             {
                 mqttPublishHeaderRO.topic = publish.topicName().asString();
-                mqttPublishHeaderRO.decodeProperties(server, publish.properties());
+                mqttPublishHeaderRO.decode(server, publish.properties());
```

Comment:
Pass in the `publish.topicName()` along with `publish.properties()` instead of modifying a field before the call?
Perhaps return the `reasonCode` directly from `decode(...)`?